lib/sync/meson.build

```diff
@@ -0,0 +1,38 @@
+libephysync_sources = [
+  'ephy-sync-crypto.c',
+  'ephy-sync-service.c',
+  'ephy-synchronizable-manager.c',
+  'ephy-synchronizable.c'
+]
+
+libephysync_deps = [
+  glib_dep,
+  gmp_dep,
+  hogweed_dep,
+  json_glib_dep,
+  libsecret_dep,
+  libsoup_dep,
+  nettle_dep,
+  webkit2gtk_dep
+]
+
+libephysync_includes = include_directories(
+  '.',
+  '..',
+  '../history',
+  '../widgets',
+  '../..',
+  '../../embed'
```

Comment:
Well this is easy, then: you can just add a helper function under lib/ to get the user agent from GSettings. Or do it manually. I think there's no need to use WebKitSettings.